### DISCUSSION TITLE
What are the recommended server specs?

### DISCUSSION THREAD

**Discussion by ksdhans** [btcpayserver/btcpayserver#6139] [General]
The BTCPay Server website lists minimum specs. But, what would you recommend to someone running a store? Hard question, I know, because it depends on how busy the server is.

I've got BTCPay Server running on an 8GB Raspberry Pi 4, and it's feeling a bit sluggish despite low traffic. Something went wrong with the bitcoin node, and resyncing is rather slow. The UI still shows "the node is starting..." even though the bitcoind log shows that it's currently syncing the blockchain. 

All of this makes me wonder if a Raspberry Pi 4 is still up to the task.

So, what hardware specs would you recommend? Anyone got experience with different servers?

**Comment by dennisreimann** [comment]
I'm running BTCPay on a Pi 4 with 4GB RAM and it keeps up with the sync. It's exposed on the internet, but receiving rather low traffic. So in general this can be done, though I wouldn't recommend that setup for eCommerce.

My recommendation for eCommerce would be a VPS, the criteria in terms of CPU and RAM depends on indivdual use case, but there are a lot of choices, they can be scaed easily and are rather cheap.

If BTCPay is run in a stationary environment as an in-store solution, the Pi setup would still work imho. If one is looking for a more performant solution with more potential to scale in the future, I'd recommend looking into mini PCs (like a Dell Optiplex).

**Comment by ksdhans** [comment]
My RPi 4 was keeping up, until something went wrong with the bitcoin node. Now it's taking a ridiculously long time just to sync the last few weeks. Strangely, the CPU usage is nowhere near 100%, and there's plenty of RAM available still. The internet connection is high-speed too. Yet, syncing is slow, as is using bitcoin-cli to fetch any info about the node (can take almost a minute). 

While using a VPS is an option, it doesn't answer the question of what server specs would be recommended.

**Reply by pavlenex** [reply]
You can use fast-sync which is recommended for RPI https://docs.btcpayserver.org/Docker/fastsync/#what-problem-does-fast-sync-solve

**Reply by ksdhans** [reply]
That page comes with some serious warnings. How safe is it? I don't have a second node to verify the blockchain with. My node has about 10.5 days worth to catch up...

BTW, what hardware would be fast enough to sync the entire blockchain in a few days? What about the "[Hack0](https://docs.btcpayserver.org/Deployment/Hack0/)" or the Beelink S12 mentioned [here](https://docs.btcpayserver.org/Deployment/LightningInABox/)?

**Reply by pavlenex** [reply]
I am unsure, I don't use any, I assume hacko, nodl and raspiblitz are good.

**Reply by ksdhans** [reply]
Ah, okay. Looks like nobody really has a good answer. 

It look like my node is periodically loosing all its peers for some reason. No idea why, yet. There's plenty of RAM available (free: 122Mi, available 3.4Gi), and CPU is hovering round 33%, and speedtest-cli reports >300Mbit/s up, & ~50 Mbit/s down.

It's still making progress with resyncing (just the last 4 hours worth left to sync), so maybe the connection dropping doesn't matter.

**Comment by ksdhans** [comment]
Finally fully resynced and up and running again. I'm going to have to monitor more closely, until we have some kind of [alert system](https://github.com/btcpayserver/btcpayserver/discussions/6138). 

I'll look into getting a more powerful server later. I'm still hoping someone with the right experience will see this discussion and comment...

**Reply by dennisreimann** [reply]
I tried to answer your initial question, but without any specifics on the use case this is hard to answer.

**Reply by ksdhans** [reply]
Yes, I know it's hard to answer without more specifics (as said in the original post). Let's just say it's for an ecommerce website running on something like WooCommerce or Shopify (so the BTCPay Server should only get traffic for payments and admin access).

What hardware specs would likely be able to handle the following while remaining reasonably responsive (so page response times within a few seconds max):
- Low traffic - up to a few sales per hour
- Medium traffic - a few sales per minute
- High traffic - a few sales per second

My RPi 4 is now back to normal, with page loads within 3-4 seconds. So, aside from something going wrong with bitcoind weeks ago and resyncing taking a long time, it's managing low traffic okay.

**Reply by dennisreimann** [reply]
For eCommerce I'd recommend running a VPS. You can go with the [options that officially support BTCPay Server](https://docs.btcpayserver.org/Deployment/webdeployment/) or do a regular Docker deployment on basically any VPS host that offers root access. 

In terms of specs I'd recommend starting with a medium option of something with 4 CPUs and 8 GB RAM. If you are aiming for something smaller, BTCPay also runs fine for me on a VPS with 2 CPUs and 4 GB RAM as well and there are also other applications running on that box. 

Diskspace depends on whether or not you want to rune a full or pruned node.

**Reply by ksdhans** [reply]
Thanks for the suggestions. I suppose that these VPS' don't indicate what the CPU specs are. I really am interested in physical server hardware specs...

**Reply by kehiy** [reply]
@dennisreimann How much disk space is required for both pruned and full nodes? if we only accept lightning, can we run btc pay server without running a full node or even a pruned node? we are aiming to use the least disk space possible. what do you suggest?

**Reply by ksdhans** [reply]
@kehiy 

I can't give you a precise number, but...

For a pruned node it depends on what size you prune the blockchain to. You can prune the blockchain storage down to 5 GB, although that's NOT recommended. 

AFAIK, the Bitcoin blockchain is currently at almost 600GB and growing ([link](https://ycharts.com/indicators/bitcoin_blockchain_size)). So the total storage needed would be about 600GB plus whatever space is needed by the OS and a BTCPay Server installation including a lightning network node. 

Of course, you'd better add a lot of spare storage to for future blockchain growth if you want to store the whole blockchain, because it's continually growing...

**Reply by kehiy** [reply]
@ksdhans Thanks for your information and help. I'm trying to run it as lightweight as possible. The number of invoices will be low at the beginning!
So we don't want to put a lot of resources on only our payment gateway!

But we can't and don't want to rely on a third party, either, so a pruned node or even free BTC RPC will be good for us.

Also, we don't have any issues with supporting only lightning!!!
